src/snapshot/benches/versionmap.rs

```diff
@@ -0,0 +1,145 @@
+    let state = Test {
+        a: vec![
+            Dummy {
+                a: "a string".to_owned(),
+                b: [0x1234u64; 32]
+            };
+            200
+        ],
+        b: 0,
+        c: 1,
+        d: 2,
+        e: vec![0x4321; 100],
+    };
+
+    let mut snapshot = Snapshot::new(vm.clone(), vm.latest_version());
+    snapshot.save(&mut snapshot_mem, &state).unwrap();
+}
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    let mut snapshot_mem = vec![0u8; 1024 * 1024 * 128];
+    let mut vm = VersionMap::new();
+
+    vm.new_version()
+        .set_type_version(Test::type_id(), 2)
+        .new_version()
+        .set_type_version(Test::type_id(), 3)
+        .new_version()
+        .set_type_version(Test::type_id(), 4)
+        .set_type_version(Dummy::type_id(), 2);
+
+    let mut slice = &mut snapshot_mem.as_mut_slice();
+    save(&mut slice, vm.clone());
+    let snapshot_len = slice.as_ptr() as usize - snapshot_mem.as_slice().as_ptr() as usize;
+    println!("Snapshot len {}", snapshot_len);
+
+    c.bench_function("Serialize in vspace=4", |b| {
+        b.iter(|| {
+            save(
+                black_box(&mut snapshot_mem.as_mut_slice()),
+                black_box(vm.clone()),
+            )
+        })
+    });
+
+    c.bench_function("Deserialize in vspace=4", |b| {
+        b.iter(|| {
+            restore(
+                black_box(&mut snapshot_mem.as_slice()),
+                black_box(vm.clone()),
+            )
+        })
+    });
+
+    // Add an extra 100 versions
+    for _ in 0..100 {
+        vm.new_version();
+    }
+
+    save(&mut snapshot_mem.as_mut_slice(), vm.clone());
+
+    c.bench_function("Serialize in vspace=100", |b| {
+        b.iter(|| {
+            save(
+                black_box(&mut snapshot_mem.as_mut_slice()),
+                black_box(vm.clone()),
+            )
+        })
+    });
+    c.bench_function("Deserialize in vspace=100", |b| {
+        b.iter(|| {
+            restore(
+                black_box(&mut snapshot_mem.as_slice()),
+                black_box(vm.clone()),
+            )
+        })
+    });
+
+    // Add an extra 100 versions
+    for _ in 0..900 {
+        vm.new_version();
+    }
+
+    // Save the snapshot at version 1001.
+    save(&mut snapshot_mem.as_mut_slice(), vm.clone());
+
+    c.bench_function("Serialize in vspace=1000", |b| {
+        b.iter(|| {
+            save(
+                black_box(&mut snapshot_mem.as_mut_slice()),
+                black_box(vm.clone()),
+            )
+        })
+    });
+    c.bench_function("Deserialize in vspace=1000", |b| {
+        b.iter(|| {
+            restore(
+                black_box(&mut snapshot_mem.as_slice()),
+                black_box(vm.clone()),
+            )
+        })
+    });
+}
+
+criterion_group! {
+    name = benches;
+    config = Criterion::default().measurement_time(std::time::Duration::from_secs(10));
```

Comment:
I ran the benches locally and for big sample sizes, I received warnings like this: 
`
Unable to complete 100 samples in 10.0s. You may wish to increase target time to 24.8s or reduce sample count to 40.
`
I think it would be nice to add a comment that says to increase this time if it's not enough or something like that or maybe a comment about why you choose `10s` here.

From what I see in the documentation: https://docs.rs/criterion/0.2.4/criterion/struct.Criterion.html#method.measurement_time, Criterion will increase automatically that time, but a comment would be useful.

Reply:
That is on purpose to lower down the run time. I will add a coment, however I will tune these  once I'll write the integration part.